porch/controllers/packagevariants/api/v1alpha1/packagevariant_types.go

```diff
@@ -77,8 +77,17 @@ type Downstream struct {
 
 // PackageVariantStatus defines the observed state of PackageVariant
 type PackageVariantStatus struct {
-	// TODO: Move this to conditions.
-	ValidationErrors []string `json:"validationErrors,omitempty"`
+	// Conditions describes the reconciliation state of the object.
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+	// DownstreamTargets contains the downstream targets that the PackageVariant
+	// either created or adopted.
+	DownstreamTargets []DownstreamTarget `json:"downstreamTargets,omitempty"`
+}
+
+type DownstreamTarget struct {
+	Name      string `json:"name,omitempty"`
+	Lifecycle string `json:"lifecycle,omitempty"`
```

Comment:
Is this the Lifecycle of the DownstreamTarget? If so, why do we need to cache it here?

Reply:
The workflow I had in mind here would be to create the PackageVariant object, then approve the draft (if it is a draft). The exact feedback was:

> So, my first question is about automation -- the PackageVariant reconciliation will create a PackageRevision as a draft. And now I want to approve the draft. What would be the correct way to wait for that PackageRevision to be created? I see nothing in the status of the PackageVariant I can wait on.
The only way I can see this work is by me knowing what the name of this downstream package is going to be and just poll to wait until it appears.

My thought was to give the client an easy way to check if the PackageRevision created needs approval. There might be cases (if the adoption policy is set to adopt already-existing packagerevisions), where you create the PackageVariant and the target package revision is already published. It isn't necessary to cache the lifecycle here though; the client could just as easily grab the name from the status and then look up the lifecycle with an api call, so I could remove it if you think that would be better.

Reply:
So creating a PackageVariant is currently a synchronous operation, so there shouldn't be any need to wait for it. However, it probably should be asynchronous (ref discussion about CRDs), so the question is still valid. 
I don't see any issues with keeping it here, we just need to make sure we verify the actual lifecycle before we try to make any changes since the cached value might be stale.

Reply:
I thought about this a bit more and I think it's better not to have the lifecycle value here - better to have the client get a fresh value from the apiserver. I'll update the PR.